packages/compose/src/hooks/use-warn-on-change/index.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal dependencies
+ */
+import usePrevious from '../use-previous';
+
+function useWarnOnChange( object, prefix = 'Change detection' ) {
```

Comment:
This is just a utility I use very often to debug performance issues. I think it can be useful for folks.

Reply:
So, to be clear, you want to add it to the compose package?

Reply:
It definitely looks useful for debugging.

Reply:
Yeah, I don't know about the compose package, I think it's fine there but not sure.

Reply:
I feel like maybe it belongs in a separate debug tools package of some sort?